service/src/test/java/com/ge/predix/acs/service/policy/admin/PolicyManagementServiceTest.java

```diff
@@ -163,6 +164,106 @@ public void testCreateApmPolicySetPositive() {
         Assert.assertEquals(this.policyService.getAllPolicySets().size(), 0);
     }
 
+    @Test
+    public void testCreatePolicyOkObligations() {
+        PolicySet policySet = this.jsonUtils.deserializeFromFile("obligation/set-with-1-policy.json", PolicySet.class);
+        String policyName = policySet.getName();
+        this.policyService.upsertPolicySet(policySet);
+        PolicySet savedPolicySet = this.policyService.getPolicySet(policyName);
+        Assert.assertEquals(2, savedPolicySet.getObligations().size());
+        this.policyService.deletePolicySet(policyName);
+        Assert.assertEquals(this.policyService.getAllPolicySets().size(), 0);
+    }
+
+    @Test(
+            expectedExceptions = { PolicyManagementException.class },
+            expectedExceptionsMessageRegExp = ".*Obligation validation failed\\. id cannot be null or blank on index.*")
```

Comment:
This test and all tests below it are annotated to use the `expectedExceptions` and `expectedExceptionsMessageRegExp` attributes which make them perfect candidates for refactoring to use an [annotation transformer](https://stackoverflow.com/a/36247029) instead of duplicating code.

Reply:
I'll still have to use the new annotation and send the regex expression to match.
I don't think that annotations can be considered  as repeated code, since it's not newly codified functionality repeated several times.